embedded-workshop-book/src/advanced-next-steps.md

```diff
@@ -0,0 +1,33 @@
+# Next Steps
+
+## String descriptors
+
+Here's one more USB exercise for you: add string descriptors support to the USB firmware.
+
+- First, you'll want to read through section 9.6.7 of the USB spec, which covers string descriptors.
+- Next, we suggest you change your *configuration* descriptor to use string descriptors. You'll want to change the `iConfiguration` field to a non-zero value.
+- Now, re-run the program to see what new control requests you get from the host.
+- You'll need to update the `usb` parser to handle the new requests.
+- Then you can extend the logic of `ep0setup` to handle these new requests.
+- Eventually, you'll need to send a string descriptor to the host. Note here that Rust string literals are UTF-8 encoded but the USB protocol uses UTF-**16** strings. You'll need to convert between these formats.
+- After you have `iConfiguration` working you can start adding strings to other descriptors like the device descriptor e.g. its `iProduct` field.
+
+To verify that string descriptors are working in a cross-platform way you can extend the `print-descs` program to also print the device's string descriptors. See the [`read_string_descriptor`] method but note that this must be called on a "device handle", which is what the commented out `open` operation does.
+
+[`read_string_descriptor`]: https://docs.rs/rusb/0.6.2/rusb/struct.DeviceHandle.html#method.read_string_descriptor
+
+## RTIC
```

Comment:
nit: I'd give this a more verbose heading, maybe something like "Explore more RTIC features"?

Reply:
good call